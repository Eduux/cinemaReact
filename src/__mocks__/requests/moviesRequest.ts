export default {
  Response: 'True',
  Search: [
    {
      Poster:
        'https://m.media-amazon.com/images/M/MV5BOTY4YjI2N2MtYmFlMC00ZjcyLTg3YjEtMDQyM2ZjYzQ5YWFkXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg',
      Title: 'Batman Begins',
      Type: 'movie',
      Year: '2005',
      imdbID: 'tt0372784',
    },
  ],
  totalResults: '405',
};
