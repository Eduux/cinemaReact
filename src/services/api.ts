import axios, { AxiosError, AxiosResponse } from 'axios';

const successResponse = ({ data }: AxiosResponse) => data;

const errorResponse = ({ response }: AxiosError) => {
  throw response;
};

const api = axios.create({
  baseURL: process.env.REACT_APP_OMDB_API,
  params: { apikey: process.env.REACT_APP_OMDB_API_KEY },
});

api.interceptors.request.use(config => config);

api.interceptors.response.use(successResponse, errorResponse);

export default api;
