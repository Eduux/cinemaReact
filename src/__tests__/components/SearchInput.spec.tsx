/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  render,
  fireEvent,
  getByPlaceholderText,
  act,
  waitFor,
} from '@testing-library/react';
import AxiosMockAdapter from 'axios-mock-adapter';

import api from 'services/api';

import Component from 'components/SearchInput';

import withMovies, { initialState } from 'containers/movies';

import moviesRequest from '__mocks__/requests/moviesRequest';

const serviceMock = new AxiosMockAdapter(api);

jest.mock('react-router-dom', () => ({
  useHistory: () => ({ push: jest.fn() }),
}));

describe('SearchInput component', () => {
  serviceMock.onGet('/').reply(200, moviesRequest);

  it('should render without data', () => {
    const Container: React.FC = withMovies(Component, initialState);

    const result = render(<Container />);

    expect(result).toMatchSnapshot();
  });

  it('should render with data', () => {
    const Container: React.FC = withMovies(Component, initialState);

    const result = render(<Container />);

    expect(result).toMatchSnapshot();
  });

  it('should render witht data and change search text', () => {
    const Container: React.FC = withMovies(Component, {
      ...initialState,
      search: 'Batman',
    });

    const result: any = render(<Container />);
    const { container } = result;

    const input: any = getByPlaceholderText(container, 'Procurar filme:');

    act(() => {
      fireEvent.change(input, { target: { value: 'Superman' } });
    });

    expect(input.value).toBe('Superman');
    expect(result).toMatchSnapshot();
  });

  it('should render witht data and change search text and submit search', () => {
    const Container: React.FC = withMovies(Component);

    const result: any = render(<Container />);
    const { container } = result;

    const input: any = getByPlaceholderText(container, 'Procurar filme:');
    act(() => {
      fireEvent.change(input, { target: { value: 'Superman' } });
      fireEvent.submit(input);
    });

    expect(input.value).toBe('Superman');
    expect(result).toMatchSnapshot();
  });
});
