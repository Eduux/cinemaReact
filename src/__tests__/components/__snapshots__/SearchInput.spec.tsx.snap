// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component should render with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDJrp fVOCov"
      >
        <form
          class="sc-hKwCoD cMFRUd"
        >
          <input
            class="sc-dkPtyc ekUMin"
            name="search"
            placeholder="Procurar filme:"
            value=""
          />
          <div
            class="sc-eCImvq NPLqQ"
          >
            <button
              class="sc-bdvvaa gWolTY"
              type="submit"
              width="90px"
            >
              Buscar
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDJrp fVOCov"
    >
      <form
        class="sc-hKwCoD cMFRUd"
      >
        <input
          class="sc-dkPtyc ekUMin"
          name="search"
          placeholder="Procurar filme:"
          value=""
        />
        <div
          class="sc-eCImvq NPLqQ"
        >
          <button
            class="sc-bdvvaa gWolTY"
            type="submit"
            width="90px"
          >
            Buscar
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchInput component should render without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDJrp fVOCov"
      >
        <form
          class="sc-hKwCoD cMFRUd"
        >
          <input
            class="sc-dkPtyc ekUMin"
            name="search"
            placeholder="Procurar filme:"
            value=""
          />
          <div
            class="sc-eCImvq NPLqQ"
          >
            <button
              class="sc-bdvvaa gWolTY"
              type="submit"
              width="90px"
            >
              Buscar
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDJrp fVOCov"
    >
      <form
        class="sc-hKwCoD cMFRUd"
      >
        <input
          class="sc-dkPtyc ekUMin"
          name="search"
          placeholder="Procurar filme:"
          value=""
        />
        <div
          class="sc-eCImvq NPLqQ"
        >
          <button
            class="sc-bdvvaa gWolTY"
            type="submit"
            width="90px"
          >
            Buscar
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchInput component should render witht data and change search text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDJrp fVOCov"
      >
        <form
          class="sc-hKwCoD cMFRUd"
        >
          <input
            class="sc-dkPtyc ekUMin"
            name="search"
            placeholder="Procurar filme:"
            value="Superman"
          />
          <div
            class="sc-eCImvq NPLqQ"
          >
            <button
              class="sc-bdvvaa gWolTY"
              type="submit"
              width="90px"
            >
              Buscar
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDJrp fVOCov"
    >
      <form
        class="sc-hKwCoD cMFRUd"
      >
        <input
          class="sc-dkPtyc ekUMin"
          name="search"
          placeholder="Procurar filme:"
          value="Superman"
        />
        <div
          class="sc-eCImvq NPLqQ"
        >
          <button
            class="sc-bdvvaa gWolTY"
            type="submit"
            width="90px"
          >
            Buscar
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchInput component should render witht data and change search text and submit search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDJrp fVOCov"
      >
        <form
          class="sc-hKwCoD cMFRUd"
        >
          <input
            class="sc-dkPtyc ekUMin"
            name="search"
            placeholder="Procurar filme:"
            value="Superman"
          />
          <div
            class="sc-eCImvq NPLqQ"
          >
            <button
              class="sc-bdvvaa gWolTY"
              type="submit"
              width="90px"
            >
              Buscar
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDJrp fVOCov"
    >
      <form
        class="sc-hKwCoD cMFRUd"
      >
        <input
          class="sc-dkPtyc ekUMin"
          name="search"
          placeholder="Procurar filme:"
          value="Superman"
        />
        <div
          class="sc-eCImvq NPLqQ"
        >
          <button
            class="sc-bdvvaa gWolTY"
            type="submit"
            width="90px"
          >
            Buscar
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
